actors/builtin/multisig/multisig_test.go

```diff
@@ -185,6 +185,182 @@ func TestPropose(t *testing.T) {
 	})
 }
 
+func TestApprove(t *testing.T) {
+	actor := msActorHarness{multisig.MultiSigActor{}, t}
+
+	receiver := newIDAddr(t, 100)
+	anne := newSecpAddr(t, 101)
+	bob := newSecpAddr(t, 102)
+	chuck := newSecpAddr(t, 103)
+
+	builder := mock.NewBuilder(context.Background(), t, receiver).WithCaller(builtin.InitActorAddr, builtin.InitActorCodeID)
+
+	t.Run("simple propose and approval", func(t *testing.T) {
+		const unlockDuration = int64(0)
+		const numApprovals = int64(2)
+		const method = int64(0)
+		const value = int64(10)
+		const txnID = int64(0)
+
+		var nilParams = abi.MethodParams{}
+		var signers = []addr.Address{anne, bob}
+
+		rt := builder.Build()
+
+		actor.mustConstructMultisigActor(rt, numApprovals, unlockDuration, signers...)
+
+		rt.SetCaller(anne, builtin.AccountActorCodeID)
+		rt.ExpectValidateCallerType(builtin.AccountActorCodeID, builtin.MultisigActorCodeID)
+		actor.mustPropose(rt, chuck, value, method, nilParams)
+
+		actor.mustContainTransactions(rt, multisig.MultiSigTransaction{
+			To:       chuck,
+			Value:    abi.NewTokenAmount(value),
+			Method:   abi.MethodNum(method),
+			Params:   nilParams,
+			Approved: []addr.Address{anne},
+		})
+
+		rt.SetCurrentBalance(abi.NewTokenAmount(value))
+		rt.SetCaller(bob, builtin.AccountActorCodeID)
+		rt.ExpectValidateCallerType(builtin.AccountActorCodeID, builtin.MultisigActorCodeID)
+		rt.ExpectSend(chuck, abi.MethodNum(method), nilParams, abi.NewTokenAmount(value), nil, 0)
+		actor.mustApprove(rt, txnID)
+
+		// Transaction should be removed from actor state after send
+		actor.mustContainTransactions(rt)
```

Comment:
Must end with `rt.Verify` to ensure any expectations (including that someone adds later) are met